VariantShredding.md

```diff
@@ -25,290 +25,319 @@
-For example, in a reconstructed Variant value, the order of object field values may be different from the original binary.
-This is allowed since the [Variant Binary Encoding](VariantEncoding.md#object-field-id-order-and-uniqueness) does not require an ordering of the field values, but the field IDs will still be ordered lexicographically according to the corresponding field names.
+Fields of an object can be shredded using a Parquet group for `typed_value` that contains shredded fields.
 
-The physical representation of scalar values may also be different in the reconstructed Variant binary.
-In particular, the [Variant Binary Encoding](VariantEncoding.md) considers all integer and decimal representations to represent a single logical type.
-This flexibility enables shredding to be applicable in more scenarios, while maintaining all information and values losslessly.
-As a result, it is valid to shred a decimal into a decimal column with a different scale, or to shred an integer as a decimal, as long as no numeric precision is lost.
-For example, it would be valid to write the value 123 to a Decimal(9, 2) column, but the value 1.234 would need to be written to the `variant_value` column.
-When reconstructing, it would be valid for a reader to reconstruct 123 as an integer, or as a Decimal(9, 2).
-Engines should not depend on the physical type of a Variant value, only the logical type.
+If the value is an object, `typed_value` must be non-null.
+If the value is not an object, `typed_value` must be null.
+Readers can assume that a value is not an object if `typed_value` is null and that `typed_value` field values are correct; that is, readers do not need to read the `value` column if `typed_value` fields satisfy the required fields.
 
-On the other hand, shredding as a different logical type is not allowed.
-For example, the integer value 123 could not be shredded to a string `typed_value` column as the string "123", since that would lose type information.
-It would need to be written to the `variant_value` column.
+Each shredded field in the `typed_value` group is represented as a required group that contains optional `value` and `typed_value` fields.
+The `value` field stores the value as Variant-encoded `binary` when the `typed_value` cannot represent the field.
+This layout enables readers to skip data based on the field statistics for `value` and `typed_value`.
 
-# Reconstructing a Variant
-
-It is possible to recover a full Variant value using a recursive algorithm, where the initial call is to `ConstructVariant` with the top-level fields, which are assumed to be null if they are not present in the schema.
+The `value` column of a partially shredded object must never contain fields represented by the Parquet columns in `typed_value` (shredded fields).
+The values stored in shredded fields must always be used when reading.
```

Comment:
What does this mean?  Do we mean:  `value` or `typed_value` inside event_type group should be used when reading event_type field?
> The values stored in shredded fields must always be used when reading.